__tests__/src/action.test.ts

```diff
@@ -18,7 +18,7 @@ jest.mock('../../src/client/vib')
 jest.spyOn(artifact, 'create')
 jest.spyOn(core, 'error')
 jest.spyOn(core, 'info')
-jest.spyOn(core, 'setFailed')
+jest.spyOn(core, 'setFailed').mockImplementation()
```

Comment:
Could you share the error you are solving? Why is this `.mockImplementation()` needed here and in other places?

Reply:
According to my previous comment: 

> It looks like there is a https://github.com/jestjs/jest/issues/14501 related to node20 and Jest that was affecting the GitHub Action, so I had to update the core.setFailed calls following this [solution](https://github.com/DataDog/synthetics-ci-github-action/commit/e1745fd59fb23cd05beabb7459223d9a7bdcbfcd). That's why there are more changes apart from the Nodejs version update

Reply:
Ok, this other link https://github.com/DataDog/synthetics-ci-github-action/pull/203 also gave me context.
There is also a `toHaveBeenCalled` to track when that mock is called.